src/storage/src/hummock/shared_buffer/shared_buffer_batch.rs

```diff
@@ -52,16 +52,24 @@ pub(crate) struct SharedBufferVersionedEntryRef<'a> {
     pub(crate) new_values: &'a [VersionedSharedBufferValue],
 }
 
+/// Return an exclusive offset of the values of key of index `i`
+fn value_end_offset<'a>(
+    i: usize,
+    entries: &'a [SharedBufferKeyEntry],
+    values: &'a [VersionedSharedBufferValue],
+) -> usize {
+    entries
+        .get(i + 1)
+        .map(|entry| entry.value_offset)
+        .unwrap_or(values.len())
+}
+
 fn values<'a>(
     i: usize,
     entries: &'a [SharedBufferKeyEntry],
     values: &'a [VersionedSharedBufferValue],
 ) -> &'a [VersionedSharedBufferValue] {
-    &values[entries[i].value_offset
-        ..entries
-            .get(i + 1)
-            .map(|entry| entry.value_offset)
-            .unwrap_or(values.len())]
+    &values[entries[i].value_offset..value_end_offset(i, entries, values)]
```

Comment:
Should we make them member functions of `SharedBufferKeyEntry` to simplify the definition? It seems they are called nowhere else besides `SharedBufferKeyEntry`.